src/benchmarks/micro/corefx/System.Threading.Tasks.Dataflow/Perf.Dataflow.cs

```diff
@@ -0,0 +1,218 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Threading.Tasks.Dataflow.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public abstract class PerfTests<T> where T : IDataflowBlock
+    {
+        protected const int MessagesCount = 100_000;
+        protected T block;
+
+        public abstract T CreateBlock();
+
+        [GlobalSetup]
+        public void BlockSetup()
+        {
+            block = CreateBlock();
+        }
```

Comment:
Why is this GlobalSetup still needed when we're calling BlockSetup on each benchmark invocation?

Reply:
We only call `BlockSetup` from the benchmark code when we are forced to call `Complete` on the tested block to have a valid test case. This concerns only two benchmarks.